src/Beman/Optional26/tests/optional_refref.t.cpp

```diff
@@ -0,0 +1,618 @@
+
+    //  SECTION("self complex")
+    {
+        EXPECT_TRUE(!(o4 == o5));
+        EXPECT_TRUE(o4 == o4);
+        EXPECT_TRUE(o4 != o5);
+        EXPECT_TRUE(!(o4 != o4));
+        EXPECT_TRUE(o4 < o5);
+        EXPECT_TRUE(!(o4 < o4));
+        EXPECT_TRUE(!(o4 > o5));
+        EXPECT_TRUE(!(o4 > o4));
+        EXPECT_TRUE(o4 <= o5);
+        EXPECT_TRUE(o4 <= o4);
+        EXPECT_TRUE(!(o4 >= o5));
+        EXPECT_TRUE(o4 >= o4);
+    }
+    //  SECTION("nullopt complex")
+    {
+        EXPECT_TRUE(!(o4 == beman::optional26::nullopt));
+        EXPECT_TRUE(!(beman::optional26::nullopt == o4));
+        EXPECT_TRUE(o4 != beman::optional26::nullopt);
+        EXPECT_TRUE(beman::optional26::nullopt != o4);
+        EXPECT_TRUE(!(o4 < beman::optional26::nullopt));
+        EXPECT_TRUE(beman::optional26::nullopt < o4);
+        EXPECT_TRUE(o4 > beman::optional26::nullopt);
+        EXPECT_TRUE(!(beman::optional26::nullopt > o4));
+        EXPECT_TRUE(!(o4 <= beman::optional26::nullopt));
+        EXPECT_TRUE(beman::optional26::nullopt <= o4);
+        EXPECT_TRUE(o4 >= beman::optional26::nullopt);
+        EXPECT_TRUE(!(beman::optional26::nullopt >= o4));
+
+        EXPECT_TRUE(o3 == beman::optional26::nullopt);
+        EXPECT_TRUE(beman::optional26::nullopt == o3);
+        EXPECT_TRUE(!(o3 != beman::optional26::nullopt));
+        EXPECT_TRUE(!(beman::optional26::nullopt != o3));
+        EXPECT_TRUE(!(o3 < beman::optional26::nullopt));
+        EXPECT_TRUE(!(beman::optional26::nullopt < o3));
+        EXPECT_TRUE(!(o3 > beman::optional26::nullopt));
+        EXPECT_TRUE(!(beman::optional26::nullopt > o3));
+        EXPECT_TRUE(o3 <= beman::optional26::nullopt);
+        EXPECT_TRUE(beman::optional26::nullopt <= o3);
+        EXPECT_TRUE(o3 >= beman::optional26::nullopt);
+        EXPECT_TRUE(beman::optional26::nullopt >= o3);
+    }
+
+    //  SECTION("with T complex")
+    {
+        EXPECT_TRUE(!(o4 == "a"));
+        EXPECT_TRUE(!("a" == o4));
+        EXPECT_TRUE(o4 != "a");
+        EXPECT_TRUE("a" != o4);
+        EXPECT_TRUE(!(o4 < "a"));
+        EXPECT_TRUE("a" < o4);
+        EXPECT_TRUE(o4 > "a");
+        EXPECT_TRUE(!("a" > o4));
+        EXPECT_TRUE(!(o4 <= "a"));
+        EXPECT_TRUE("a" <= o4);
+        EXPECT_TRUE(o4 >= "a");
+        EXPECT_TRUE(!("a" >= o4));
+
+        EXPECT_TRUE(o4 == "hello");
+        EXPECT_TRUE("hello" == o4);
+        EXPECT_TRUE(!(o4 != "hello"));
+        EXPECT_TRUE(!("hello" != o4));
+        EXPECT_TRUE(!(o4 < "hello"));
+        EXPECT_TRUE(!("hello" < o4));
+        EXPECT_TRUE(!(o4 > "hello"));
+        EXPECT_TRUE(!("hello" > o4));
+        EXPECT_TRUE(o4 <= "hello");
+        EXPECT_TRUE("hello" <= o4);
+        EXPECT_TRUE(o4 >= "hello");
+        EXPECT_TRUE("hello" >= o4);
+    }
+}
+
+TEST(OptionalRefRefTest, Triviality) {
+    EXPECT_TRUE(std::is_trivially_copy_constructible<beman::optional26::optional<int&&>>::value);
```

Comment:
Maybe use an alias `using Type = beman::optional26::optional<int&&>;` and then `EXPECT_TRUE(std::is_trivially_copy_constructible<Type>::value`);

Btw, we can put a follow-up commit/PR to remove duplication across `*.t.cpp`. We can create a new issue. for that.